File: language-support/js/daml-ledger/src/validation/command.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from "..";
import { union, Validation } from "./base";
import { CreateCommand } from "./create_command";
import { ExerciseCommand } from "./exercise_command";

function values(): Record<keyof ledger.Command, Validation> {
    return {
        create: CreateCommand,
        exercise: ExerciseCommand
    };
}

export const Command: Validation = union('Command', values);

File: canton/community/app/src/main/scala/com/digitalasset/canton/cli/Command.scala
// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.canton.cli

import java.io.File

sealed trait Command {}

object Command {

  /** Run the process as a server (rather than an interactive repl)
    */
  object Daemon extends Command

  /** Run a console script then close
    *
    * @param scriptPath the path to the script
    */
  final case class RunScript(scriptPath: File) extends Command

  final case class Generate(target: Generate.Target) extends Command

  object Generate {
    sealed trait Target

    object RemoteConfig extends Target
  }
}
